internal/targetedretries/substitution.go

```diff
@@ -23,6 +23,23 @@ var (
 	keywordRegexp     = regexp.MustCompile(`^{{\s?(\w+)\s?}}$`)
 )
 
+var SubstitutionsByFramework = map[v1.Framework]Substitution{
+	v1.DotNetxUnitFramework:          new(DotNetxUnitSubstitution),
+	v1.ElixirExUnitFramework:         new(ElixirExUnitSubstitution),
+	v1.GoGinkgoFramework:             new(GoGinkgoSubstitution),
+	v1.GoTestFramework:               new(GoTestSubstitution),
+	v1.JavaScriptCypressFramework:    new(JavaScriptCypressSubstitution),
+	v1.JavaScriptJestFramework:       new(JavaScriptJestSubstitution),
+	v1.JavaScriptMochaFramework:      new(JavaScriptMochaSubstitution),
+	v1.JavaScriptPlaywrightFramework: new(JavaScriptPlaywrightSubstitution),
+	v1.PHPUnitFramework:              new(PHPUnitSubstitution),
+	v1.PythonPytestFramework:         new(PythonPytestSubstitution),
+	v1.PythonUnitTestFramework:       new(PythonUnitTestSubstitution),
+	v1.RubyCucumberFramework:         new(RubyCucumberSubstitution),
+	v1.RubyMinitestFramework:         new(RubyMinitestSubstitution),
+	v1.RubyRSpecFramework:            new(RubyRSpecSubstitution),
+}
```

Comment:
This isn't run-specific, so I moved it here to be shared by whatever needs retry substitutions.